examples/colors/main.rs

```diff
@@ -0,0 +1,194 @@
+        Default::default(),
+    );
+
+    application.connect_activate(build_ui);
+    application.run();
+}
+
+fn build_ui(app: &gtk::Application) {
+    let window = gtk::ApplicationWindow::builder()
+        .default_width(600)
+        .default_height(400)
+        .application(app)
+        .title("Colors")
+        .build();
+
+    let header = gtk::HeaderBar::new();
+    window.set_titlebar(Some(&header));
+
+    let model = ColorStore::new(8);
+
+    let sort_model = gtk::SortListModel::new(Some(&model), None::<&gtk::Sorter>);
+
+    let selection = gtk::MultiSelection::new(Some(&sort_model));
+
+    let factory = gtk::SignalListItemFactory::new();
+    factory.connect_setup(move |_, list_item| {
+        let picture = gtk::Picture::new();
+        picture.set_size_request(32, 32);
+        list_item.set_child(Some(&picture));
+    });
+
+    factory.connect_bind2(|_, object: gtk::gdk::Paintable, picture: gtk::Picture| {
+        picture.set_paintable(Some(&object));
+    });
+
+    let list = [
+        "8", "64", "512", "4096", "32768", "262144", "2097152", "16777216",
+    ];
+    let selected = Rc::new(RefCell::new(4096u32));
+
+    let refill_btn = gtk::Button::with_mnemonic("_Refill");
+    refill_btn.connect_clicked(clone!(@strong model, @strong selected => move |_btn| {
+        println!("TODO!");
+    }));
+    header.pack_start(&refill_btn);
+
+    let progress = gtk::ProgressBar::new();
+    progress.set_hexpand(true);
+    progress.set_valign(gtk::Align::Start);
+
+    sort_model.connect_pending_notify(clone!(@strong progress => move |model| {
+        let total = model.n_items();
+        let total = total.max(1); /* avoid div by 0 below */
+        let pending = model.pending();
+
+        progress.set_visible(pending != 0);
+        progress.set_fraction((total - pending) as f64 / total as f64);
+    }));
+
+    let label = gtk::Label::new(Some("0 /"));
+    let attrs = gtk::pango::AttrList::new();
+    attrs.insert(gtk::pango::AttrFontFeatures::new("tnum").upcast());
+    label.set_attributes(Some(&attrs));
+    label.set_xalign(1.0);
+    header.pack_start(&label);
+
+    let dropdown = gtk::DropDown::from_strings(&list);
+    dropdown.connect_selected_notify(clone!(@strong selected, @strong label => move |dropdown| {
+        let num_str = list[dropdown.selected() as usize];
+        if let Ok(num) = num_str.parse::<u32>() {
+            model.set_property("size", num);
+            label.set_label(&format!("{} /", num_str));
+
+            *selected.borrow_mut() = num;
+        }
+    }));
+    dropdown.set_selected(3);
+    header.pack_start(&dropdown);
+
+    let list = [
+        "Unsorted",
+        "Red",
+        "Green",
+        "Blue",
+        "Hue",
+        "Saturation",
+        "Value",
+    ];
+    let dropdown = gtk::DropDown::from_strings(&list);
+    dropdown.connect_selected_notify(clone!(@strong selected => move |dropdown| {
+        match list[dropdown.selected() as usize] {
```

Comment:
this block of code could be simplified a lot; just make the match return the `Some(property_name)` and create the sorter once instead of duplicating the same code again and again